warehouse/accounts/security_policy.py

```diff
@@ -184,8 +184,7 @@ def _permits_for_user_policy(acl, request, context, permission):
     if (
         isinstance(res, Allowed)
         and not request.identity.has_primary_verified_email
-        and request.matched_route.name.startswith("manage")
-        and request.matched_route.name != "manage.account"
+        and not request.matched_route.name.startswith("manage.unverified-account")
```

Comment:
Allow the user to click the verification link

```suggestion
        and not request.matched_route.name == "accounts.verify-email"
        and not request.matched_route.name.startswith("manage.unverified-account")
```

Reply:
Fixed in 5085f78, 168233f and e23d0cc.